dockerfiles/theia/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM node:8-alpine
+# build dependencies requireed to compile a custom Theia
+RUN apk add --no-cache make gcc g++ python git openssh bash
+WORKDIR /home/theia
+# build Theia with all extensions to persist yarn cache in the image and
+# have default Theia build in the workspace in case no plugins are requested
+ADD theia-full-package.json /home/theia/package.json
```

Comment:
it looks like there is a fork of upstream package.json, how to be in sync with upstream changes automatically ?

Reply:
I think we can have two options here:

1. curl original Theia package.json as a Dockerfile instruction
2. Rely on our CI to do this after a triggered webhook.

Reply:
having the github URL of theia package.json will allow to only re-run the RUN instruction only if the remote side is updated

Reply:
(so we can reuse layer cache if there is no remote change)